From oneukum@suse.de Wed May  2 01:33:27 2007
From: Oliver Neukum <oneukum@suse.de>
Date: Wed, 2 May 2007 10:33:10 +0200
Subject: USB: fix omninet memory leak found by coverity
To: Eric Sesterhenn / Snakebyte <snakebyte@gmx.de>, linux-usb-devel@lists.sourceforge.net, "Greg Kroah-Hartman" <gregkh@suse.de>
Message-ID: <200705021033.10521.oneukum@suse.de>
Content-Disposition: inline


the transfer of allocating the descriptor in attach and no longer in open
was incomplete resulting in a memory leak coverity spotted. This fix
is against the patch set you posted.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/omninet.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/usb/serial/omninet.c
+++ b/drivers/usb/serial/omninet.c
@@ -165,12 +165,10 @@ static int omninet_open (struct usb_seri
 {
 	struct usb_serial	*serial = port->serial;
 	struct usb_serial_port	*wport;
-	struct omninet_data	*od = usb_get_serial_port_data(port);
 	int			result = 0;
 
 	dbg("%s - port %d", __FUNCTION__, port->number);
 
-	od = kmalloc( sizeof(struct omninet_data), GFP_KERNEL );
 	wport = serial->port[1];
 	wport->tty = port->tty;
 
